tests/config/test_functional_toml_parsing.py

```diff
@@ -0,0 +1,51 @@
+from pathlib import Path
+from typing import Tuple
+from unittest.mock import patch
+
+import pytest
+
+from pylint import run_pylint
+
+HERE = Path(__file__).parent
+USER_SPECIFIC = str(HERE.parent.parent)
+FUNCTIONAL_DIR = HERE / "functional"
+# We use string then recast to path so we can use -k in pytest otherwise
+# we get 'pyproject_toml_path0' as a test name
+TOML_PATHS = [str(p) for p in (FUNCTIONAL_DIR).rglob("*.toml")]
+FILE_TO_LINT = str(HERE / "file_to_lint.py")
+
+
+def get_relpath(path: str) -> str:
+    """Get the relative path we want without user specific information"""
+    return path.replace(USER_SPECIFIC, "")[1:]
+
+
+def get_expected(pyproject_toml_path: str) -> Tuple[int, str]:
+    expected_output = ""
+    expected_code = 0
+    expected_result_path = Path(pyproject_toml_path + ".expected")
+    if expected_result_path.exists():
+        expected_code = 2
+        with open(expected_result_path, encoding="utf8") as f:
+            expected_output = f.read()
+        expected_output = expected_output.format(
+            abspath=pyproject_toml_path, relpath=get_relpath(pyproject_toml_path)
+        )
+    return expected_code, expected_output
+
+
+@pytest.mark.parametrize("pyproject_toml_path", TOML_PATHS)
+def test_config_pyproject_toml(pyproject_toml_path, capsys):
+    """Functional tests for configurations."""
+    msg = f"Wrong result with configuration {pyproject_toml_path}"
+    expected_code, expected_output = get_expected(pyproject_toml_path)
+    with patch(
+        "sys.argv", ["pylint", "--rcfile", pyproject_toml_path, FILE_TO_LINT, "-r", "n"]
+    ):
+        with pytest.raises(SystemExit) as exc_info:
+            run_pylint()
+        assert exc_info.value.code == expected_code, msg
+        out, err = capsys.readouterr()
+        # rstrip() applied so we can have a final newline in the expected test file
+        assert expected_output.rstrip() == out.rstrip(), msg
+        assert not err, msg
```

Comment:
I like this! At some point we might want to make this accessible for other tests as well, but for this PR I think this works fine!

Reply:
Yes, I made that for toml file, only but it could be any configuration file... We'll just have to select file not ending in ".expected".